kernel/thread.c

```diff
@@ -922,5 +922,29 @@ static inline k_ticks_t z_vrfy_k_thread_timeout_expires_ticks(
 	Z_OOPS(Z_SYSCALL_OBJ(t, K_OBJ_THREAD));
 	return z_impl_k_thread_timeout_expires_ticks(t);
 }
+
+void z_impl_sys_cache_flush(void *addr, size_t size)
```

Comment:
why are those being added in kernel/thread.c? How is this related to threads?

Reply:
Was just a place to start with. Have shifted this code to a new file kernel/cache_handlers.c.